src/test/java/no/unit/nva/database/UserDbTest.java

```diff
@@ -0,0 +1,189 @@
+        dynamoFunctionalityTestUser.setUsername(SOME_USERNAME);
+        assertThat(dynamoFunctionalityTestUser.getUsername(), is(equalTo(SOME_USERNAME)));
+    }
+
+    @Test
+    public void getUsernameShouldGetTheSetUsernameToUserObject() throws InvalidUserException {
+        assertThat(dynamoFunctionalityTestUser.getUsername(), is(nullValue()));
+
+        dynamoFunctionalityTestUser.setUsername(SOME_USERNAME);
+        assertThat(dynamoFunctionalityTestUser.getUsername(), is(equalTo(SOME_USERNAME)));
+    }
+
+    @Test
+    public void getTypeShouldReturnConstantTypeValue() {
+        assertThat(dynamoFunctionalityTestUser.getType(), is(equalTo(UserDb.TYPE)));
+    }
+
+    @Test
+    public void setTypeShouldNotChangeTheReturnedTypeValue() {
+        dynamoFunctionalityTestUser.setType("NotExpectedType");
+        assertThat(dynamoFunctionalityTestUser.getType(), is(equalTo(UserDb.TYPE)));
+    }
+
+    @Test
+    public void getHashKeyKeyShouldReturnTypeAndUsernameConcatenation()  {
+        String expectedHashKey = String.join(UserDb.FIELD_DELIMITER, UserDb.TYPE, SOME_USERNAME);
+        assertThat(sampleUser.getPrimaryHashKey(), is(equalTo(expectedHashKey)));
+    }
+
+    @Test
+    public void builderShouldSetTheHashKeyBasedOnusername() throws InvalidUserException {
+        sampleUser.setPrimaryHashKey("SomeOtherHashKey");
+        String expectedHashKey = String.join(UserDb.FIELD_DELIMITER, UserDb.TYPE, SOME_USERNAME);
+        assertThat(sampleUser.getPrimaryHashKey(), is(equalTo(expectedHashKey)));
+    }
+
+    @Test
+    public void userDbShouldBeWriteableToDatabase() throws InvalidUserException {
+        DynamoDBMapper mapper = clientToLocalDatabase();
+        assertDoesNotThrow(() -> mapper.save(sampleUser));
+    }
+
+    @Test
+    public void userDbShouldBeReadFromDatabaseWithoutDataLoss() throws InvalidUserException {
+        UserDb insertedUser = UserDb.newBuilder()
+            .withUsername(SOME_USERNAME)
+            .withInstitution(SOME_INSTITUTION)
+            .withRoles(SAMPLE_ROLES)
+            .build();
+        DynamoDBMapper mapper = clientToLocalDatabase();
+        mapper.save(insertedUser);
+        assertThatNoPublicFieldIsNull(insertedUser);
+        UserDb savedUser = mapper.load(UserDb.class, insertedUser.getPrimaryHashKey());
+        assertThat(savedUser, is(equalTo(insertedUser)));
+    }
+
+    @Test
+    public void builderShouldThrowExceptionWhenAtLeastOneInputRoleIsInvalid() throws InvalidUserException {
+        List<RoleDb> roles = SAMPLE_ROLES;
+        RoleDb invalidRole = new RoleDb();
+        invalidRole.setName(BLANK_STRING);
+        roles.add(invalidRole);
+
+        UserDb insertedUser = UserDb.newBuilder()
+            .withUsername(SOME_USERNAME)
+            .withInstitution(SOME_INSTITUTION)
+            .withRoles(roles)
+            .build();
+    }
+
+
+    private void assertThatNoPublicFieldIsNull(UserDb insertedUser) {
```

Comment:
Maybe we could implement the hamcrest Matcher interface

Reply:
I cannot fix this now, because code does not compile, because it is a big PR broken into pieces. I will create an issue as a note